### PR TITLE
Bump Kubernetes to v1.30.0

### PR DESCRIPTION
**Pull request by dgrisonnet**
Bump all the dependencies to latest including Kubernetes to 1.30.0 and golang to 1.22.2.

While doing the bump I noticed that we inadvertently added some unused openapi files in https://github.com/kubernetes-sigs/custom-metrics-apiserver/commit/759711cd34faa23f85c8afa587f6344e8203fa72, so I cleaned them up.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,23 +1,23 @@
 module sigs.k8s.io/custom-metrics-apiserver
 
-go 1.21
+go 1.22.0
```
Comment (CatherineF-dev) [inline go.mod:3]: nit: 1.22.1

https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#go

Comment (dgrisonnet) [reply go.mod]: Fixed

**Pull request merged**